lib/src/rules/missing_conditional_import.dart

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2021, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+
+import '../analyzer.dart';
+
+const _desc = r'Missing conditional import.';
+
+const _details = r'''
+
+**DON'T** reference files that do not exist in conditional imports.
+
+Code may fail at runtime if the condition evaluates such that the missing file
+needs to be imported.
+
+**BAD:**
+```dart
+import 'file_that_does_not_exist.dart';
+```
+
+**GOOD:**
+```dart
+import 'file_that_exists.dart';
+```
+
+''';
+
+class MissingConditionalImport extends LintRule {
+  MissingConditionalImport()
+      : super(
+            name: 'missing_conditional_import',
```

Comment:
As for naming, I might suggest that we use a name similar to the error code (`URI_DOES_NOT_EXIST`) such as `conditional_uri_does_not_exist`. There's a companion error code in the analyzer (`URI_HAS_NOT_BEEN_GENERATED`) and we might want to have two separate messages for this lint to distinguish between those cases, but that's less clear to me.

(The only convention that I'm aware of for naming lints is that we're now actively discouraging using 'avoid', 'prefer' and other such designations from the style guide because those tend to change over time leaving us with misleading lint names. Other than that, lints should be named to either express the condition they're trying to ensure or the condition they're trying to prevent.)

Reply:
Thanks! I've changed it to `conditional_uri_does_not_exist`.

> There's a companion error code in the analyzer (URI_HAS_NOT_BEEN_GENERATED) and we might want to have two separate messages for this lint to distinguish between those cases, but that's less clear to me.

I don't know much about that so have no opinion, but if you think it would also be useful here I'm happy to add it - it doesn't seem like a complicated change (just changing the code/error depending on whether the uri ends with [these suffixes](https://github.com/dart-lang/sdk/blob/e995cb5f7cd67d39c1ee4bdbe95c8241db36725f/pkg/analyzer/lib/src/util/file_paths.dart#L75)?).

Reply:
Thinking about it a bit more, I'd propose that we leave it as is. If we get feedback from users that a different message would be helpful we can always update it later.